segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/SegmentRollingPolicy.java

```diff
@@ -16,7 +16,10 @@
  * A generic rolling policy that can be applied to any Storage unit.
  */
 public final class SegmentRollingPolicy {
-    public static final SegmentRollingPolicy NO_ROLLING = new SegmentRollingPolicy(Long.MAX_VALUE);
+    /**
+     * Max rolling length is 2^62 so that we can use CompactLong in serialization everywhere.
+     */
+    public static final SegmentRollingPolicy NO_ROLLING = new SegmentRollingPolicy(Long.MAX_VALUE / 4 );
```

Comment:
what is the impact of changing this value ?
is there any backward compatibility concern ?

Reply:
For new installations that shouldn't be problem. For migration we will have to  set it to new  max rolling length. From practical point of view max rolling length will now be 2305843009213693951 bytes which is still really big.

Reply:
For serialization we want to use compact long representation. If not we'll require 8 bytes for each length related field.  Using compact serialization allows us to save size of records. However compact can handle up to 2^62.  Long.MAX_VALUE is an arbitrary limit there is no real reason the default value has to be that high. With 2^62 the new limit is 2305843009213693951 which is still really big default value,